packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -931,6 +944,49 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     await this._txPool.updatePendingAndQueued();
   }
 
+  public async setAccountBalance(
+    address: Address,
+    newBalance: BN
+  ): Promise<void> {
+    const account = await this._stateManager.getAccount(address);
+    account.balance = newBalance;
+    await this._stateManager.putAccount(address, account);
+    this._irregularStatesByBlockNum.set(
```

Comment:
I considered this but concluded that the answer was no, because I thought that the World State Trie only includes balances, but now that you're asking the question I'm realizing that I was not seeing past the oversimplification of the diagram [here](https://ethereum.stackexchange.com/a/6413/6802).  I will set in the next three functions too, and also add tests to verify this behavior.

Reply:
Validated in https://github.com/nomiclabs/hardhat/pull/1435/commits/4e84f59dcdf3f06faa294f84be93d90c926929eb (see [CI result](https://github.com/nomiclabs/hardhat/runs/2775774777#step:9:6879)) and addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/bed37937ee1009830cc3400acee2d7ece1359e57